cpp/ql/lib/semmle/code/cpp/ir/internal/IRUtilities.qll

```diff
@@ -30,7 +39,10 @@ Type getVariableType(Variable v) {
         result = v.getInitializer().getExpr().getType()
         or
         not exists(v.getInitializer()) and result = v.getType()
-      else result = v.getType()
+      else
+        if isNonReferenceStructuredBinding(v)
+        then exists(ReferenceType r | r.getBaseType() = v.getUnderlyingType() | result = r)
```

Comment:
I'm not sure whether this `exists` will always work. Also I'm not sure whether to use `v.getUnderlyingType()` or `v.getUnspecifiedType()`. I did already see I cannot use `v.getType()` as that doesn't work in the case of `typedefs`.

Reply:
This question is still relevant, although the code here has changed slightly.

Reply:
Changed to `getUnspecifiedType`.